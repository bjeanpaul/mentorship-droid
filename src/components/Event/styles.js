import { StyleSheet } from 'react-native';
import colors from 'src/constants/colors';
import { FONT, FONT_WEIGHT } from 'src/constants/styles';

export default StyleSheet.create({
  container: {
    backgroundColor: colors.EVENT_BG,
    margin: 24,
    marginTop: 23,
    paddingTop: 20,
    paddingBottom: 20,
    paddingLeft: 16,
    paddingRight: 8,
    borderRadius: 4,
    justifyContent: 'space-between',
    alignItems: 'center',
    flexDirection: 'row',
  },
  icon: {
    width: 56,
    height: 56,
    borderRadius: 56,
    marginRight: 16,
    backgroundColor: colors.EVENT_ICON_BG,
  },
  iconImage: {
    width: 56,
    height: 56,
    borderRadius: 56,
  },
  textContainer: {
    flex: 1,
    paddingTop: 3,
    flexDirection: 'column',
    alignSelf: 'flex-start',
  },
  title: {
    fontSize: 16,
    color: colors.EVENT_TITLE,
    textAlign: 'left',
    fontFamily: FONT.BOLD,
    fontWeight: FONT_WEIGHT.BOLD,
  },
  blurb: {
    fontSize: 15,
    color: colors.EVENT_BLURB,
    textAlign: 'left',
  },
  more: {
    alignItems: 'center',
    justifyContent: 'center',
  },
});
