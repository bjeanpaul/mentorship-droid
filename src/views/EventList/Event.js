import React, { PropTypes } from 'react';

import containers from 'src/containers/events';


const Event = ({
  event,
}) => {
  const { eventType } = event;
  const EventContainer = containers[eventType];
  return EventContainer && <EventContainer {...event} />;
};


Event.propTypes = {
  event: PropTypes.shape({
    eventType: PropTypes.string.isRequired,
  }).isRequired,
};


export default Event;
