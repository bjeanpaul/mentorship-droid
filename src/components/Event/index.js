import moment from 'moment';
import React, { PropTypes } from 'react';
import { View, Image, TouchableNativeFeedback } from 'react-native';
import { Text } from 'src/components';
import styles from 'src/components/Event/styles';
import images from 'src/constants/images';

const Event = ({
  date,
  icon,
  title,
  blurb,
  onPress,
}) => (
  <TouchableNativeFeedback onPress={() => onPress && onPress()}>
    <View style={styles.container}>
      <View style={styles.icon}>
        <Image style={styles.iconImage} source={icon} />
      </View>

      <View style={styles.textContainer}>
        <Text style={styles.title}>{title}</Text>

        <Text numberOfLines={2} style={styles.blurb}>
          {blurb || moment(date).format('dddd Do, MMMM YYYY')}
        </Text>
      </View>

      {
        onPress && <View style={styles.moreContainer}>
          <Image
            style={styles.more}
            source={images.EVENT_MORE}
          />
        </View>
      }
    </View>
  </TouchableNativeFeedback>
);


Event.propTypes = {
  date: PropTypes.string.isRequired,
  icon: PropTypes.any.isRequired,
  title: PropTypes.string.isRequired,
  blurb: PropTypes.string,
  onPress: PropTypes.func,
};


export default Event;
