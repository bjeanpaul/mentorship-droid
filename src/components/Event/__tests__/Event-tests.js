import React from 'react';
import { noop } from 'lodash';

import Event from 'src/components/Event';


describe('Event', () => {
  const createComponent = (props = {}) => (
    <Event
      date="2016-09-16T11:19:17.368442Z"
      icon={23}
      title="pew pew pew"
      blurb="i am a blurb"
      {...props}
    />
  );

  it('should render the props', () => {
    const el = render(createComponent());
    expect(el.toJSON()).toMatchSnapshot();
  });

  it('should render a formatted date if the blurb is undefined', () => {
    const el = render(createComponent({
      date: '2016-09-16T11:19:17.368442Z',
      blurb: void 0,
    }));

    expect(el.toJSON()).toMatchSnapshot();
  });

  it('should render with a "more" indicator if onPress is given', () => {
    const el = render(createComponent({
      onPress: noop,
    }));

    expect(el.toJSON()).toMatchSnapshot();
  });

  it('should render a counter if a count is given', () => {
    const el = render(createComponent({
      count: 23,
    }));

    expect(el.toJSON()).toMatchSnapshot();
  });

  it('should call onPress when pressed', () => {
    const onPress = jest.fn();

    shallow(createComponent({ onPress }))
      .simulate('press');

    expect(onPress.mock.calls).toEqual([[]]);
  });
});
