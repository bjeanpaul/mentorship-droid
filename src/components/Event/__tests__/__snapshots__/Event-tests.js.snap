exports[`Event should render a formatted date if the blurb is undefined 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "#fff",
      "borderRadius": 4,
      "flexDirection": "row",
      "margin": 24,
      "marginTop": 0,
      "padding": 20,
      "paddingLeft": 16,
      "paddingRight": 16
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "#999",
        "borderRadius": 28,
        "height": 56,
        "marginRight": 16,
        "width": 56
      }
    }>
    <Image
      source={1} />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column"
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 13,
            "textAlign": "center"
          },
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text_bold",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "left"
          }
        ]
      }>
      pew pew pew
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 13,
            "textAlign": "center"
          },
          Object {
            "color": "#80979a",
            "fontSize": 15,
            "textAlign": "left"
          }
        ]
      }>
      Friday 16th, September 2016
    </Text>
  </View>
</View>
`;

exports[`Event should render the props 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "#fff",
      "borderRadius": 4,
      "flexDirection": "row",
      "margin": 24,
      "marginTop": 0,
      "padding": 20,
      "paddingLeft": 16,
      "paddingRight": 16
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "#999",
        "borderRadius": 28,
        "height": 56,
        "marginRight": 16,
        "width": 56
      }
    }>
    <Image
      source={1} />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column"
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 13,
            "textAlign": "center"
          },
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text_bold",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "left"
          }
        ]
      }>
      pew pew pew
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 13,
            "textAlign": "center"
          },
          Object {
            "color": "#80979a",
            "fontSize": 15,
            "textAlign": "left"
          }
        ]
      }>
      i am a blurb
    </Text>
  </View>
</View>
`;
